connect/cli/plugins/product/sync/translations.py

```diff
@@ -0,0 +1,157 @@
+            errors.append('Translation ID is required to update or delete a translation')
+
+        if (
+            data.action in ('update', 'create')
+            and data.autotranslation not in ('Enabled', 'Disabled')
+        ):
+            errors.append(
+                'Autotranslation must be `Enabled` or `Disabled`. '
+                f'Provided {data.autotranslation}',
+            )
+
+        if data.action == 'create' and not data.locale:
+            errors.append('Locale is required to create a translation')
+
+        return errors
+
+    def _process_rows_data(self, rows_data):
+        self._translations_autotranslating = []
+        # for a consistent sync first delete translations, then update and create
+        for row_idx, data in rows_data['delete'].items():
+            self._handle_action(self._action_delete, row_idx, data)
+
+        for row_idx, data in rows_data['update'].items():
+            self._handle_action(self._action_update, row_idx, data)
+
+        if len(rows_data['create']) > 0:
+            # if there are any translations to create, then the context_id is needed
+            try:
+                ctx = self._client.ns('localization').contexts.filter(
+                    instance_id=self._product_id,
+                ).first()
+                for row_idx, data in rows_data['create'].items():
+                    self._handle_action(partial(self._action_create, ctx['id']), row_idx, data)
+            except ClientError as e:
+                self._mstats.error(str(e).split('\n'))
+
+    def _handle_action(self, action_handler, row_idx, data):
+        try:
+            translation = action_handler(data)
+            if translation:
+                fill_translation_row(self._ws, row_idx, translation, update_mode=True)
+        except Exception as e:
+            self._mstats.error(str(e).split('\n'), row_idx)
+
+    def _action_delete(self, data):
+        self._set_process_description(f'Deleting translation {data.translation_id}')
+        try:
+            self._client.ns('localization').translations[data.translation_id].delete()
+        except ClientError as e:
+            if e.status_code != 404:
+                raise
+        self._mstats.deleted()
+
+    def _action_update(self, data):
+        self._set_process_description(f'Updating translation {data.translation_id}')
+        payload = {
+            'description': data.description or "",
+            'auto': {
+                'enabled': data.autotranslation == 'Enabled',
+            },
+        }
+        translation = self._client.ns('localization').translations[data.translation_id].update(payload)
+        self._mstats.updated()
+        if translation['auto']['enabled']:
+            self._translations_autotranslating.append(translation['id'])
+        return translation
+
+    def _action_create(self, context_id, data):
+        self._set_process_description(f'Creating translation {data.translation_id}')
```

Comment:
ops, well seen 👍 ... yes, in fact it will show whatever is in the excel "Translation ID" column, which is ignored for the 'create' action. I will set it to just "Creating translation...". I will do it on master and then update here.